libsolidity/AST.h

```diff
@@ -147,6 +147,67 @@ class Declaration: public ASTNode
 	Declaration const* m_scope;
 };
 
+
+/**
+* Generic Parameter description used by @see FunctionDescription to return
+* a descripton of its parameters.
+*/
+struct ParamDescription
+{
+	ParamDescription(std::string const& _name, std::string const& _type):
+		m_description(_name, _type){}
+
+	bool operator!=(ParamDescription const& _other) const;
+	std::ostream& operator<<(std::ostream& os) const;
+
+	std::string getName() const;
+	std::string getType() const;
+
+	std::pair<std::string, std::string> m_description;
+};
+
+
+/**
+ * Generic function description able to describe both normal functions and
+ * functions that should be made as accessors to state variables
+ */
+struct FunctionDescription
```

Comment:
As I said this was a work in progress. Plan to simplify the accessors of the struct and remove those not needed.